### PR TITLE
Add convenience SqlService.executeUpdate method

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Simply add a convenience method that takes care of closing underlying SqlResult when no real result is expected (non-DQL statement). In such case closing is not strictly necessary but is good practice since SqlResult is AutoCloseable.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases
- [x] New public APIs have `@Nonnull/@Nullable` annotations
- [x] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/sql/SqlService.java [hunk 1]
```diff
@@ -89,6 +89,42 @@ default SqlResult execute(@Nonnull String sql, Object... arguments) {
...
+        SqlStatement statement = new SqlStatement(sql);
+
+        if (arguments != null) {
+            for (Object arg : arguments) {
+                statement.addParameter(arg);
+            }
+
+        }
+        try (SqlResult ignored = execute(statement)) {
+        }
```
Comment (k-jamroz) [inline hazelcast/src/main/java/com/hazelcast/sql/SqlService.java:125]:
```suggestion
        execute(statement).close();
```

File: hazelcast/src/main/java/com/hazelcast/sql/SqlService.java [hunk 2]
```diff
@@ -89,6 +89,42 @@ default SqlResult execute(@Nonnull String sql, Object... arguments) {
...
+     * @throws IllegalArgumentException if the SQL string is empty
+     * @throws HazelcastSqlException    in case of execution error
+     * @see SqlService
+     * @see SqlStatement
+     * @see #execute(SqlStatement)
+     *
+     * @since 5.3
+     */
+    @SuppressWarnings({"EmptyTryBlock", "checkstyle:EmptyBlock"})
+    default void executeStatement(@Nonnull String sql, Object... arguments) {
```
Comment (k-jamroz) [inline hazelcast/src/main/java/com/hazelcast/sql/SqlService.java:115]: I do not like the name

File: hazelcast/src/main/java/com/hazelcast/sql/SqlService.java (hunk 1 above)
Comment (k-jamroz) [reply hazelcast/src/main/java/com/hazelcast/sql/SqlService.java:125]: btw, maybe we should ensure that there were really no rows.

**Comment by k-jamroz** [comment]
I was thinking about adding a similar helper but for scalar queries (1 row with 1 column): https://github.com/hazelcast/hazelcast/pull/23140/files#diff-ae948b2b986ea1c9536e8266a88d84a2c31535f5eca06ff372520046b62b4df4R106

I was worried if it will not make the API less clear and harder to use. Same question here.

**Comment by TomaszGaweda** [comment]
Queries other than DQL are quite common, so I think it's convenient to have such method. We use such pattern a lot in our tests and have a lot of warnings because of that :)

File: hazelcast/src/main/java/com/hazelcast/sql/SqlService.java (hunk 2 above)
Comment (TomaszGaweda) [reply hazelcast/src/main/java/com/hazelcast/sql/SqlService.java:115]: I mean, even if it does, those will be simply discarded. Side effects in SQLa are rare but also possible.

I don't have an idea for better name. Maybe `runCommand`?

**Comment by k-jamroz** [comment]
> Queries other than DQL are quite common, so I think it's convenient to have such method. We use such pattern a lot in our tests and have a lot of warnings because of that :)

Such method is basically equivalent to `sqlClient.execute("...").close()`. I think it might be useful, but what is your opinion about bloating the API?

File: hazelcast/src/main/java/com/hazelcast/sql/SqlService.java [hunk 3]
```diff
@@ -89,6 +89,39 @@ default SqlResult execute(@Nonnull String sql, Object... arguments) {
         return execute(statement);
     }
 
+    /**
+     * Convenient method to execute a distributed query with the given
```
Comment (k-jamroz) [inline hazelcast/src/main/java/com/hazelcast/sql/SqlService.java:93]: "Convenient" -> "Convenience"

We should mention that it executed not any query, but only queries with do not return rows in the beginning of the description.

**Review by Fly-Style — APPROVED** [review]

**Pull request merged**